src/Fetch.js

```diff
@@ -0,0 +1,530 @@
+  }
+}
+
+function __emscripten_fetch_xhr(fetch, onsuccess, onerror, onprogress) {
+  var url = HEAPU32[fetch + 8 >> 2];//{{{ makeGetValue('fetch', 8/*TODO:structs_info.jsonify this*/, 'i32') }}};
+  if (!url) {
+#if FETCH_DEBUG
+    console.error('fetch: XHR failed, no URL specified!');
+#endif
+    onerror(fetch, 0, 'no url specified!');
+    return;
+  }
+  var url_ = Pointer_stringify(url);
+
+  var fetch_attr = fetch + 112/*TODO:structs_info*/;
+  var requestMethod = Pointer_stringify(fetch_attr);
+  if (!requestMethod) requestMethod = 'GET';
+  var userData = HEAPU32[fetch_attr + 32 >> 2];//{{{ makeGetValue('fetch_attr', 32/*TODO:structs_info.jsonify this*/, 'i32') }}};
+  var fetchAttributes = HEAPU32[fetch_attr + 48 >> 2];//{{{ makeGetValue('fetch_attr', 48/*TODO*/, 'i32') }}};
+  var timeoutMsecs = HEAPU32[fetch_attr + 52 >> 2];//{{{ makeGetValue('fetch_attr', 52/*TODO*/, 'i32') }}};
+  var withCredentials = !!HEAPU32[fetch_attr + 56 >> 2];//({{{ makeGetValue('fetch_attr', 56/*TODO*/, 'i32') }}});
+  var destinationPath = HEAPU32[fetch_attr + 64 >> 2];//{{{ makeGetValue('fetch_attr', 64/*TODO*/, 'i32') }}};
+  var userName = HEAPU32[fetch_attr + 68 >> 2];//{{{ makeGetValue('fetch_attr', 68/*TODO*/, 'i32') }}};
+  var password = HEAPU32[fetch_attr + 72 >> 2];//{{{ makeGetValue('fetch_attr', 72/*TODO*/, 'i32') }}};
+  var requestHeaders = HEAPU32[fetch_attr + 76 >> 2];//{{{ makeGetValue('fetch_attr', 76/*TODO*/, 'i32') }}};
+  var overriddenMimeType = HEAPU32[fetch_attr + 80 >> 2];//{{{ makeGetValue('fetch_attr', 80/*TODO*/, 'i32') }}};
+
+  var fetchAttrLoadToMemory = !!(fetchAttributes & 1/*EMSCRIPTEN_FETCH_LOAD_TO_MEMORY*/);
+  var fetchAttrStreamData = !!(fetchAttributes & 2/*EMSCRIPTEN_FETCH_STREAM_DATA*/);
+  var fetchAttrPersistFile = !!(fetchAttributes & 4/*EMSCRIPTEN_FETCH_PERSIST_FILE*/);
+  var fetchAttrAppend = !!(fetchAttributes & 8/*EMSCRIPTEN_FETCH_APPEND*/);
+  var fetchAttrReplace = !!(fetchAttributes & 16/*EMSCRIPTEN_FETCH_REPLACE*/);
+  var fetchAttrNoDownload = !!(fetchAttributes & 32/*EMSCRIPTEN_FETCH_NO_DOWNLOAD*/);
+  var fetchAttrSynchronous = !!(fetchAttributes & 64/*EMSCRIPTEN_FETCH_SYNCHRONOUS*/);
+  var fetchAttrWaitable = !!(fetchAttributes & 128/*EMSCRIPTEN_FETCH_WAITABLE*/);
+
+  var userNameStr = userName ? Pointer_stringify(userName) : undefined;
+  var passwordStr = password ? Pointer_stringify(password) : undefined;
+  var overriddenMimeTypeStr = overriddenMimeType ? Pointer_stringify(overriddenMimeType) : undefined;
+
+  var xhr = new XMLHttpRequest();
+  xhr.timeout = timeoutMsecs;
+  xhr.withCredentials = withCredentials;
+#if FETCH_DEBUG
+  console.log('fetch: xhr.timeout: ' + xhr.timeout + ', xhr.withCredentials: ' + xhr.withCredentials);
+  console.log('fetch: xhr.open(requestMethod="' + requestMethod + '", url: "' + url_ +'", userName: ' + userNameStr + ', password: ' + passwordStr + ');');
+#endif
+  xhr.open(requestMethod, url_, !fetchAttrSynchronous, userNameStr, passwordStr);
```

Comment:
Thanks, this is documented in https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/timeout, adapted for that.